src/parser.rs

```diff
@@ -134,8 +74,13 @@ pub fn parse(mut lexer: ParseSession) -> PResult<ParsedAst> {
                 }
             }
             KeywordEndActions | End => return Ok((unit, lexer.diagnostics)),
-            Error => return Err(unidentified_token(&lexer)),
-            _ => return Err(unexpected_token(&lexer)),
+            _ => {
+                return Err(Diagnostic::unexpected_token_found(
+                    None,
```

Comment:
This is expecting a block starting keyword (PROGRAM, Function...)
We could probably add that as expected.

I will act like I did not see the EndActions/End in the line above I think EndActions should have already been handeled anyway we should reconsider why it's here.